pfr/engine/etl_eventlog.py

```diff
@@ -0,0 +1,98 @@
+
+    def _get_case_edge(self, sublog):
+        sublog[Eventlog.activity_next.name] = sublog[Eventlog.activity.name].shift(-1)
+        sublog[Eventlog.transition_start_ts.name] = sublog[Eventlog.activity_end_ts.name]
+        value_for_NaT = sublog.iloc[-1].at[Eventlog.transition_start_ts.name]
+        sublog[Eventlog.transition_end_ts.name] = (sublog[Eventlog.activity_start_ts.name]
+                                                   .shift(-1, fill_value=value_for_NaT))
+        return sublog
+
+    def create(self, raw_eventlog):
+        timestamp_mask = self._config["timestampMask"]
+        eventlog = pd.DataFrame()
+        eventlog[Eventlog.case_id.name] = raw_eventlog[self._config['eventlogInputColumns']['caseId']].astype(str)
+        eventlog[Eventlog.resource.name] = raw_eventlog[self._config['eventlogInputColumns']['resource']]
+        eventlog[Eventlog.activity.name] = raw_eventlog[self._config['eventlogInputColumns']['activity']]
+        eventlog[Eventlog.activity.name] = eventlog.apply(lambda row: trim_if_string(row[Eventlog.activity.name]), axis=1)
+        eventlog[Eventlog.id.name] = range(0, len(eventlog))
+        eventlog[Eventlog.activity_end_ts.name] = raw_eventlog[self._config['eventlogInputColumns']['endTimestamp']]
+        eventlog[Eventlog.activity_end_ts.name] = pd.to_datetime(eventlog[Eventlog.activity_end_ts.name],
+                                                            errors='raise',
+                                                            format=timestamp_mask,
+                                                            utc=True).dt.tz_localize(None)
+        if "startTimestamp" in self._config['eventlogInputColumns'].keys():
+            eventlog[Eventlog.activity_start_ts.name] = pd.to_datetime(
+                raw_eventlog[self._config['eventlogInputColumns']['startTimestamp']],
+                errors='raise',
+                format=timestamp_mask,
+                utc=True).dt.tz_localize(None)
+        else:
+            eventlog[Eventlog.activity_start_ts.name] = eventlog[Eventlog.activity_end_ts.name]
+
+        eventlog[Eventlog.activity_business_duration_h.name] = eventlog.apply(
+            lambda row: get_business_durtation_h(row[Eventlog.activity_start_ts.name], row[Eventlog.activity_end_ts.name], self._businesshrs),
+            axis=1)
+        eventlog[Eventlog.activity_duration_h.name] = eventlog.apply(
+            lambda row: get_duration_h(row[Eventlog.activity_start_ts.name], row[Eventlog.activity_end_ts.name]), axis=1)
+        eventlog[Eventlog.activity_start_date.name] = eventlog[Eventlog.activity_start_ts.name].dt.date.astype("datetime64[ns]")
+        eventlog[Eventlog.activity_end_date.name] = eventlog[Eventlog.activity_end_ts.name].dt.date.astype("datetime64[ns]")
+        eventlog.sort_values(by=[Eventlog.case_id.name, Eventlog.activity_start_ts.name], inplace=True)
+        eventlog = self._add_fake_activities(eventlog)
+        eventlog = (eventlog
```

Comment:
When running the example I see following warning message:
```
 ./pfutil -c example_data/p2p/config.yaml  put -e example_data/p2p/eventlog.csv --csv-out output
/home/damian/projects/processifier/pfutil/pfr/engine/etl_eventlog.py:86: FutureWarning: Not prepending group keys to the result index of transform-like apply. In the future, the group keys will be included in the index, regardless of whether the applied function returns a like-indexed object.
To preserve the previous behavior, use

	>>> .groupby(..., group_keys=False)

To adopt the future behavior and silence this warning, use 

	>>> .groupby(..., group_keys=True)
  eventlog = (eventlog
```
That must be fixed.

+ Some informative messages should be printed to the user (minimum full paths to the created files)